### ISSUE TITLE
nodata value being ignored for NaN

### ISSUE THREAD

**Issue by cerolinx** [cogeotiff/rio-tiler#690]
The #672 do not work with numpy NaN.
numpy.float32('nan') == numpy.float32('nan') and numpy.float64('nan') == numpy.float64('nan') are evaluated False.

Proposal (Line 240 in rio-tiler/rio_tiler/reader.py):
```
            # if data has Nodata then we simply make sure the mask == the nodata
            if nodata is not None:
                if numpy.isnan(nodata):
                    data.mask = numpy.isnan(data.data)
                else:
                    data.mask = data.data == nodata
```
The tile form my float32 dataset with NaN as nodata before:

![tile_b](https://github.com/cogeotiff/rio-tiler/assets/12711846/34e08902-4726-4752-afb5-0736bec244f1)
 
And after:

![tile](https://github.com/cogeotiff/rio-tiler/assets/12711846/06f3d1f8-3e22-4892-8a13-499297d569ce)

Thanks!

**Comment by vincentsarago** [comment]
thanks for the report @cerolinx 🙏 

would you be able to submit a PR (with tests)? 

Thanks

**Comment by vincentsarago** [comment]
closed in #691
